distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -303,8 +324,39 @@ type Vector
              [1, 2, 3, 4, 5].filter (> 3)
     filter : (Any -> Boolean) -> Vector Any
     filter predicate =
-        check acc ix = if predicate ix then acc + [ix] else acc
-        this.fold [] check
+        builder = here.new_builder
+        acc = this.fold Nothing acc-> elem->
```

Comment:
This pattern deserves a linear abstraction IMO. Something like:
```
Vector.Builder.unsafe_append = <CURRENT APPEND IMPL>

Vector.Builder.append : a ! e -> Vector.Builder ! e
Vector.Builder.append a = case a of
    _ ->
        this.unsafe_append a
        this
```
And then the implementation of this becomes
```
builder = this.fold here.new_builder bldr-> elem->
    if predicate elem then bldr.append elem else bldr
builder.to_vector
```

Reply:
Implemented it.
Thanks for the awesome suggestion, that seems to be the right way to go, indeed.